Python/bytecodes.c

```diff
@@ -782,6 +869,26 @@ dummy_func(
             DISPATCH_INLINED(gen_frame);
         }
 
+        inst(INSTRUMENTED_YIELD_VALUE, (retval -- unused)) {
+            assert(frame != &entry_frame);
+            PyGenObject *gen = _PyFrame_GetGenerator(frame);
+            gen->gi_frame_state = FRAME_SUSPENDED;
+            _PyFrame_SetStackPointer(frame, stack_pointer - 1);
+            int err = _Py_call_instrumentation_arg(
+                    tstate, PY_MONITORING_EVENT_PY_YIELD,
+                    frame, next_instr-1, retval);
+            if (err) goto error;
+            tstate->exc_info = gen->gi_exc_state.previous_item;
```

Comment:
I just noticed this here -- many INSTRUMENTED opcodes have a tail that's a copy of the corresponding base opcode. Maybe that should be done using a goto and a label instead? It causes code bloat and risks the two versions getting out of sync, and also requires future maintainers to manually compare the two versions to see how they differ.

Reply:
We could implement the instrumented instructions as macro instructions, e.g.:
```
macro(YIELD_VALUE) = YIELD_VALUE_START + YIELD_VALUE_END;
macro(INSTRUMENTED_YIELD_VALUE) = YIELD_VALUE_START + MONITOR_YIELD_VALUE + YIELD_VALUE_END;
```

I was reluctant to do so earlier, as the code generator was under active development and it might have needed changes to the code generator which I didn't want to manage at the same time.

Any objection to me doing that in another PR once this one is merged?

Reply:
Sure, please add a TODO to the issue to remind us.

Reply:
Done